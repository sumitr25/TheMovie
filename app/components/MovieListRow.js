import React from 'react';
import { Image, StyleSheet, Text, TouchableOpacity, View } from 'react-native';
import colors from '../res/colors';
import fonts from '../res/fonts';

const MovieListRow = ({ item, onRowPress }) => {
  return (
    <TouchableOpacity
      onPress={() => {
        onRowPress(item);
      }}>
      <View style={styles.container}>
        <View style={styles.imageContainer}>
          <Image
            style={styles.logo}
            source={{
              uri: 'https://reactnative.dev/img/tiny_logo.png',
            }}
          />
        </View>
        <View style={styles.detailContainer}>
          <Text style={styles.title}>Title</Text>
          <Text style={styles.description}>description</Text>
        </View>
      </View>
    </TouchableOpacity>
  );
};

const styles = StyleSheet.create({
  container: {
    flex: 1,
    backgroundColor: colors.white,
    flexDirection: 'row',
    paddingHorizontal: fonts.normal,
    paddingVertical: fonts.larger,
    borderBottomWidth: 0.2,
  },
  imageContainer: {
    backgroundColor: colors.white,
  },
  detailContainer: {
    backgroundColor: colors.white,
    paddingLeft: fonts.larger,
  },
  title: {
    color: colors.black,
    fontWeight: fonts.bold,
    fontSize: fonts.normal,
  },
  description: {
    color: colors.black,
  },
  logo: {
    width: 66,
    height: 66,
  },
});

export default MovieListRow;
