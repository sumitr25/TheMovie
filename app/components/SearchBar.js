import { debounce } from 'lodash';
import React, { useCallback, useState } from 'react';
import {
  Image,
  StyleSheet,
  TextInput,
  TouchableOpacity,
  View
} from 'react-native';
import colors from '../res/colors';
import fonts from '../res/fonts';
import images from '../res/images';
import strings from '../res/strings';

const SearchBar = ({ onTextChange }) => {
  const [searchString, setSearchString] = useState('');

  const request = debounce(searchText => {
    onTextChange(searchText);
  }, 500);

  const debounceRequest = useCallback(searchText => request(searchText), []);

  const onSearchTextChange = searchText => {
    setSearchString(searchText);
    debounceRequest(searchText);
  };

  const onCrossButtonPressed = () => {
    setSearchString('');
    onTextChange(strings.movie.movieList.loadingKeyword);
  };

  return (
    <View style={styles.container}>
      <Image style={styles.searchIcon} source={images.searchIcon} />
      <TextInput
        style={styles.searchBox}
        placeholder={strings.movie.movieList.search}
        placeholderTextColor={colors.black}
        value={searchString}
        onChangeText={query => {
          onSearchTextChange(query);
        }}
      />
      {searchString.length > 0 ? (
        <TouchableOpacity
          onPress={() => {
            onCrossButtonPressed();
          }}>
          <Image style={styles.crossIcon} source={images.crossIcon} />
        </TouchableOpacity>
      ) : null}
    </View>
  );
};

const styles = StyleSheet.create({
  container: {
    flexDirection: 'row',
    width: '100%',
    paddingVertical: fonts.mini,
    paddingHorizontal: fonts.small,
    fontSize: fonts.large,
    backgroundColor: colors.silver,
    alignItems: 'center',
  },
  searchIcon: {
    width: fonts.largest,
    height: fonts.largest,
  },
  crossIcon: {
    width: fonts.larger,
    height: fonts.larger,
  },
  searchBox: {
    color: colors.black,
    paddingLeft: fonts.mini,
    fontSize: fonts.large,
    flex: 1,
  },
});

export default SearchBar;
