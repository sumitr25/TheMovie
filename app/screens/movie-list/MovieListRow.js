import React, { useState } from 'react';
import { Image, StyleSheet, Text, TouchableOpacity, View } from 'react-native';
import { colors } from '../../res/colors';
import { dimensions } from '../../res/dimensions';
import { fontWeight } from '../../res/fontWeight';
import { images } from '../../res/images';
import { strings } from '../../res/strings';
import { getImageUri } from '../../services/network/Manager';

const MovieListRow = ({ item, onRowPress }) => {
  const [isImageLoading, setIsImageLoading] = useState(false);

  return (
    <TouchableOpacity onPress={() => onRowPress(item)}>
      <View style={styles.container}>
        <View style={styles.imageContainer}>
          <Image
            style={styles.movieThumbnail}
            source={
              isImageLoading
                ? { uri: getImageUri(item.poster_path) }
                : images.defaultImage
            }
            onLoad={() => setIsImageLoading(true)}
            defaultSource={images.defaultImage}
            resizeMode={'cover'}
          />
        </View>
        <View style={styles.detailContainer}>
          <Text style={styles.title} numberOfLines={1}>
            {item.title}
          </Text>
          <KeyInfo
            keyValue={strings.movie.movieList.releaseDate}
            value={item.release_date}
          />
          <KeyInfo
            keyValue={strings.movie.movieList.ratings}
            value={item.vote_average}
          />
        </View>
      </View>
    </TouchableOpacity>
  );
};

const KeyInfo = ({ keyValue, value }) => {
  return (
    <View style={styles.keyInfo}>
      <Text style={styles.keyLabel}>{keyValue}:</Text>
      <Text style={styles.valueLabel}>{value}</Text>
    </View>
  );
};

const styles = StyleSheet.create({
  container: {
    flex: 1,
    backgroundColor: colors.white,
    flexDirection: 'row',
    paddingVertical: dimensions.larger,
    borderBottomWidth: dimensions.borderBottomWidth,
  },
  imageContainer: {
    backgroundColor: colors.white,
  },
  detailContainer: {
    backgroundColor: colors.white,
    paddingLeft: dimensions.larger,
    paddingRight: dimensions.xLarge,
  },
  title: {
    color: colors.black,
    fontWeight: fontWeight.bold,
    fontSize: dimensions.large,
  },
  description: {
    color: colors.black,
    paddingRight: dimensions.large,
  },
  movieThumbnail: {
    width: dimensions.movieThumbnail,
    height: dimensions.movieThumbnail,
    borderRadius: dimensions.mini,
  },
  keyInfo: {
    flexDirection: 'row',
    marginBottom: dimensions.mini,
  },
  keyLabel: {
    color: colors.darkGray,
  },
  valueLabel: {
    color: colors.black,
    marginLeft: dimensions.mini,
  },
});

export default MovieListRow;
