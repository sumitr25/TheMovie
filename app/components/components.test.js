import React from 'react';
import {shallow} from 'enzyme';
import SearchBar from './SearchBar';
import MovieListRow from './MovieListRow';

describe('SearchBar', () => {
    describe('Rendering', () => {
        it('should match to snapshot', () => {
            const component = shallow(<SearchBar onTextChange={() => {}} />)
            expect(component).toMatchSnapshot()
        });
    });
});

describe('MovieListRow', () => {
    describe('Rendering', () => {
        it('should match to snapshot', () => {
            const component = shallow(<MovieListRow
                onRowPress={() => {}}
                item={{}}
                />)
            expect(component).toMatchSnapshot()
        });
    });
});
