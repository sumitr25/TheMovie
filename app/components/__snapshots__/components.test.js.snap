// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MovieListRow Rendering should match to snapshot 1`] = `ShallowWrapper {}`;

exports[`SearchBar Rendering should match to snapshot 1`] = `ShallowWrapper {}`;
